query/physicalplan/aggregate.go

```diff
@@ -127,6 +126,11 @@ func (a *HashAggregate) SetNextCallback(nextCallback func(r arrow.Record) error)
 	a.nextCallback = nextCallback
 }
 
+// Go translation of boost's hash_combine function.
+func hashCombine(lhs, rhs uint64) uint64 {
+	return lhs ^ (rhs + 0x9e3779b9 + (lhs << 6) + (lhs >> 2))
```

Comment:
Where does this `0x9e3779b9` come from? Is that part of the boot's hash_combine?